packages/utils/src/__tests__/joiner/remote-joiner-data.ts

```diff
@@ -0,0 +1,466 @@
+              id: 101,
+            },
+          },
+          {
+            product_id: 101,
+            variant_id: 992,
+            quantity: 5,
+            product: {
+              name: "Product 1",
+              id: 101,
+            },
+          },
+        ],
+        user_id: 4,
+        user: {
+          fullname: "444 Doe full name",
+          email: "444444@example.com",
+          products: [
+            {
+              id: 4,
+              product_id: 103,
+              product: {
+                name: "Product 3",
+                id: 103,
+              },
+            },
+          ],
+          id: 4,
+        },
+      },
+      {
+        number: "ORD-202",
+        date: "2023-04-01T12:00:00Z",
+        products: [
+          {
+            product_id: [101, 103],
+            variant_id: 993,
+            quantity: 4,
+            product: [
+              {
+                name: "Product 1",
+                id: 101,
+              },
+              {
+                name: "Product 3",
+                id: 103,
+              },
+            ],
+          },
+        ],
+        user_id: 1,
+        user: {
+          fullname: "John Doe full name",
+          email: "johndoe@example.com",
+          products: [
+            {
+              id: 1,
+              product_id: 102,
+              product: {
+                name: "Product 2",
+                id: 102,
+              },
+            },
+          ],
+          id: 1,
+        },
+      },
+    ])
+  })
+
+  it("Query a service expanding an inverse relation", async () => {
+    let query: RemoteJoinerQuery = {
+      service: "Variant",
+      fields: ["name"],
+      expands: [
+        {
+          property: "orders",
+          fields: ["order"],
+        },
+        {
+          property: "orders.order",
+          fields: ["id", "number"],
+          args: [
+            {
+              name: "limit",
+              value: "5",
+            },
+          ],
+        },
+        {
+          property: "orders.order.products.product",
+          fields: ["name"],
+        },
+        {
+          property: "orders.variant",
+          fields: ["name"],
+        },
+      ],
+    }
+
+    query = RemoteJoiner.parseQuery(`
+      query {
+        variant {
+          id
+          name
+          orders {
+            order {
+              number
+              products {
+                product {
+                  name
+                }
+                variant {
+                  name
+                }
+              }
+            }
+          }
+        }
+      }
+    `)
+    const data = await joiner.query(query)
+
+    expect(data).toEqual([
+      {
+        id: 991,
+        name: "Product variant 1",
+        orders: {
+          order: {
+            number: "ORD-001",
+            products: [
+              {
+                product_id: 101,
+                variant_id: 991,
+                quantity: 1,
```

Comment:
Good question, the `quantity` seems to be a bug with the mocked data and properties that the service has to resolve in one call. I will work on a fix, but at least it isn't a blocker at the moment.

The other fields, specially ids, are the ones needed to link the entities, and to avoid do that magically, and make it easier to debug at least at the beginning, we're bringing these fields as well